### ISSUE TITLE
Closing client does not flush metrics

### ISSUE THREAD

**Issue by fcobb** [DataDog/datadog-go#138]
Hello,

I am observing no metrics being sent after upgrading to version 3.4.1 from 2.3.0.

I wrote a simple test to reproduce this behaviour and I am using Wireshark to observe the traffic I send. I can see via Wireshark that on version 3.4.1. the metrics are not being sent, whereas they are on 2.3.0.

If on version 3.4.1 I add a `client.Flush()` (as commented out in the below test) the metrics are sent. I expected the closing of the client to flush the metrics, as seen in the following lines
https://github.com/DataDog/datadog-go/blob/master/statsd/statsd.go#L568-L569'

Is this to be expected?

Go version: 1.14.0

Thanks,
Fraser

```
import (
	"testing"

	"github.com/DataDog/datadog-go/statsd"
)

func Test_BasicMetric(t *testing.T){
	client, err := statsd.New("127.0.0.1:8125")
	if err != nil {
		t.Fatalf("failed to create client: %s", err)
	}

	if err := client.Count("name", 1, []string{"tag"}, 1); err != nil {
		t.Fatalf("failed to send metric: %s", err)
	}

	// _ = client.Flush()

	if err := client.Close(); err != nil {
		t.Fatalf("failed to close client: %s", err)
	}
}
```

**Comment by mrwonko** [comment]
We noticed the same thing upon upgrading from 2.2.0 to 3.5.0. I don't think this is expected, since there's explicitly

```Go
// Finally flush any remaining metrics that may have come in at the last moment
```

in the `Close()` function.

**Comment by nightlyone** [comment]
The sequence in https://github.com/DataDog/datadog-go/blob/f137255/statsd/sender.go#L113-L117
seems to be the culprit. 

After the final flush the loop could still be active and the queue channel as well as the stop could both be ready and select will use a random one.

I would suggest doing something like this instead
```go
	s.stop <- struct{}{}
	<-s.stop
	s.flush()
	return s.transport.Close()
```
which would stop the sender in all cases, reliable does the final flush and also closes the transport at a point after all IO has been done and where no further IO can be scheduled anymore.

**Comment by hush-hush** [comment]
Hi @fcobb,

Thanks for reporting this issue. @nightlyone is right about the source of the bug. I opened a PR fixing the issue https://github.com/DataDog/datadog-go/pull/144 which should fix the issue.

**Comment by hush-hush** [comment]
The PR #144 has been merged and will be included in the next release `v3.5.0` (most likely release this week).

Closing this issue but feel free to reopen it or a new one if needed
